components/raftstore/src/store/entry_storage.rs

```diff
@@ -248,8 +253,12 @@ impl EntryCache {
                 }
                 break;
             }
-            let (_, dangle_size) = cached_entries.take_entries();
-            mem_size_change -= dangle_size as i64;
+            // keep the applying entries if not all of them are persisted.
```

Comment:
Pls add more comments about this change.
```suggestion
            // Do not evict cached entries if not all of them are persisted.
            // Following the changes in #16834, there is a possibility that applying
            // entries are not yet fully persisted. Therefore, it should not free these
            // entries until they are completely persisted.
```